### PR TITLE
Order fulfill mutation adjustments

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because it modifies `orderFulfill` mutation to fit new fulfillments confirmation feature.

Todo:
* [x] Raise an error if order is unpaid and `fulfillment_allow_unpaid` is False
* [x] Created fulfillments should have status `WAITING_FOR_ACCEPTANCE` if `fulfillment_auto_confirm` is False
* [x] When `fulfillment_auto_confirm` is False, stocks cannot be decreased
* [x] New event when fulfillment waiting for accept is created
* [x] DO NOT trigger `FULFILLMENT_CREATED` and `ORDER_FULFILLED` if fulfillment awaits acceptance, otherwise trigger it.

⚠️ This is a PR to feature branch.

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [x] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3840 Update orderFulfill mutation](https://app.clickup.com/t/2549495/SALEOR-3840)

**Review by IKarbowiak — APPROVED** [review]

File: saleor/order/events.py [hunk 1]
```diff
@@ -454,6 +454,19 @@ def fulfillment_fulfilled_items_event(
     )
 
 
+def fulfillment_awaits_acceptance_event(
+    *, order: Order, user: UserType, fulfillment_lines: List[FulfillmentLine]
```
Comment (maarcingebala) [inline saleor/order/events.py:458]: This should support also `app` users?

File: saleor/order/actions.py [hunk 2]
```diff
@@ -213,6 +213,22 @@ def order_fulfilled(
...
+    fulfillment_lines: List["FulfillmentLine"],
+    manager: "PluginsManager",
+    _notify_customer=True,
+):
+    order = fulfillments[0].order
+    update_order_status(order)
+    events.fulfillment_awaits_acceptance_event(
+        order=order, user=user, fulfillment_lines=fulfillment_lines
+    )
+    manager.order_updated(order)
```
Comment (fowczarek) [inline saleor/order/actions.py:229]: It's should be triggered in `on_commit` of this transaction.

File: saleor/order/events.py (hunk 1 above)
Comment (tomaszszymanski129) [reply saleor/order/events.py]: Added app support 👍🏻

**Review by maarcingebala — APPROVED** [review]

File: saleor/order/actions.py [hunk 3]
```diff
@@ -497,6 +515,7 @@ def create_fulfillments(
     fulfillment_lines_for_warehouses: Dict,
     manager: "PluginsManager",
     notify_customer: bool = True,
+    confirmed: bool = True,
```
[the remark this answers is not in the document]
Comment (tomaszszymanski129) [reply saleor/order/actions.py]: Decided to go with `approved` / `approval` / `approve` in feature scope.

**Comment by maarcingebala** [comment]
LGTM 👍

**Review by fowczarek — APPROVED** [review]

**Pull request merged**